File: aoc-2021/rust/src/day1.rs
use std::fs;
use itertools::Itertools;

pub fn part_one() {
    let res = fs::read_to_string("../input/day1.txt")
        .unwrap()
        .split("\n")
        .map(|x| x.parse::<i32>().unwrap())
        .tuple_windows::<(i32, i32)>()
        .fold(0, |acc, (prev, next)| if prev < next { acc + 1 } else { acc });
    println!("{}", res)
}
pub fn part_two() {
    let res = fs::read_to_string("../input/day1.txt")
        .unwrap()
        .split("\n")
        .map(|x| x.parse::<i32>().unwrap())
        .tuple_windows::<(i32, i32, i32)>()
        .map(|(a, b, c)| a + b + c)
        .tuple_windows::<(i32, i32)>()
        .fold(0, |acc, (prev, next)| if prev < next { acc + 1 } else { acc });
    println!("{}", res)
}

File: aoc-2021/python/day1.py
nums = [int(x) for x in open('../input/day1.txt').readlines()]

def part_one():
    count = 0
    for i, num in enumerate(nums[:-1]):
        if num < nums[i+1]:
            count += 1
    print(count) 

def part_two():
    count = 0
    for i in range(2, len(nums)):
        prev = sum(nums[i-2:i+1])
        curr = sum(nums[i-1:i+2])
        if prev < curr:
            count += 1
    print(count) 

# part_one()
part_two()